src/bokeh/util/serialization.py

```diff
@@ -144,7 +144,8 @@ def convert_date_to_datetime(obj: dt.date) -> float:
         datetime
 
     '''
-    return (dt.datetime(*obj.timetuple()[:6], tzinfo=dt.timezone.utc) - DT_EPOCH).total_seconds() * 1000
+    dt_obj = dt.datetime(obj.year, obj.month, obj.day, tzinfo=dt.timezone.utc)
+    return (dt_obj - DT_EPOCH).total_seconds() * 1000
```

Comment:
```suggestion
    return (dt.datetime.combine(obj, dt.time(), tzinfo=dt.timezone.utc) - DT_EPOCH).total_seconds() * 1000
```

The suggested change is not the fastest version I found. To me it looks like `dt.datetime.combine()` is faster.

Here is a code snipped to validate:

```python
import datetime as dt

DT_EPOCH = dt.datetime.fromtimestamp(0, tz=dt.timezone.utc)
test_date = dt.date.today()

def convert_date_to_datetime(obj: dt.date) -> float:
    return (dt.datetime(*obj.timetuple()[:6], tzinfo=dt.timezone.utc) - DT_EPOCH).total_seconds() * 1000

def convert_date_to_datetime_suggested(obj: dt.date) -> float:
    dt_obj = dt.datetime(obj.year, obj.month, obj.day, tzinfo=dt.timezone.utc)
    return (dt_obj - DT_EPOCH).total_seconds() * 1000

def convert_date_to_datetime_combine(obj: dt.date) -> float:
    return (dt.datetime.combine(obj, dt.time(), tzinfo=dt.timezone.utc) - DT_EPOCH).total_seconds() * 1000
```

And the results on my machine:
<img width="671" height="422" alt="grafik" src="https://github.com/user-attachments/assets/c9ae1cd5-646f-4e44-b94d-d313fa57af65" />

Reply:
This makes sense to me, `dt.combine` is dispatches directly to a C implementation so several python-level attribute lookups are avoided. As long as we are engaging in micro-optimizations we might as well do them as good as possible....

@misrasaurabh1 I will however be honest and say that I am starting to question the cost/benefit of these optimizations relative to the developer effort spent reviewing them. Many Bokeh programs will never use this codepath. Call me old-fashioned but I think it might be better to come at optimization from the other direction, i.e. run real typical code examples and identify hotspots in codepaths that are known to be *common*.

Reply:
@bryevdv sounds good to me. Can you provide me some benchmarks to optimize? Codeflash gets better results when it has a benchmark to optimize.

Reply:
I can try finding optimizations for `bokeh/examples` if that's the right set of perf benchmarks to target

Reply:
Most of the things under `examples/basic`, `examples/output`, `examples/styling`, and `examples/topics` would be a reasonable place to look for things closer to "real world" code.